packages/http-client-csharp/generator/docs/backward-compatibility.md

```diff
@@ -0,0 +1,187 @@
+# Backward Compatibility Support
+
+## Table of Contents
+
+- [Overview](#overview)
+- [How It Works](#how-it-works)
+- [Supported Scenarios](#supported-scenarios)
+  - [Model Factory Methods](#model-factory-methods)
+  - [Model Properties](#model-properties)
+  - [API Version Enum](#api-version-enum)
+
+## Overview
+
+The TypeSpec C# generator supports backward compatibility (back-compat) to ensure that existing client code continues to work when the service API evolves. This is achieved by comparing the current generated code against a previous contract (assembly) and automatically generating compatibility shims where needed.
+
+The backward compatibility system uses the `LastContractView` mechanism to access information about previously generated types, methods, and properties from the last released version of the library.
+
+## How It Works
+
+When generating code, the generator can optionally receive a compiled assembly from the previous version of the library (the "last contract"). The generator:
+
+1. Analyzes the current TypeSpec specification and generates types based on the current API
+2. Compares the generated types against the types in the last contract
+3. Automatically generates compatibility methods, properties, or enum members where differences are detected
+
+## Supported Scenarios
+
+### Model Factory Methods
+
+Model factory methods are public static methods that enable creating model instances for mocking and testing purposes. The generator attempts to maintain backward compatibility for these methods across API changes.
+
+#### Scenario 1: New Model Property Added
```

Comment:
nit: I see there is only a single scenario for Model Properties/Api Version Enum, but it is a little strange that they are numbered here but not in the others. I would just remove the "N" from Scenario N.